src/30_Advanced/SeatMap.html

```diff
@@ -0,0 +1,179 @@
+    decide whether to show a + or - button.
+      * `selectedSeats` variables keeps a  list of selected seat ids.  -->
+    <amp-state id="seatmap">
+      <script type="application/json">
+        {
+          "scale": 1,
+          "selectedSeats": [],
+          "zoom": 0
+        }
+      </script>
+    </amp-state>
+
+    <!-- 
+    We combine `amp-list`, `amp-bind` and `amp-pan-zoom` to implement a seatmap:
+
+    * `amp-list` is used to keep the availability of the seatmap up to date.
+    * `amp-bind` is used to update the status of the seats: when a seat is selected or unselected we update the color and 
+    we add the seat id to the `seatmap.selectedSeats` variable. 
+    `amp-pan-zoom` is used to make possible to pan & zoom on mobile devices.
+    We are using the `transformEnd` action from [amp-pan-zoom](https://www.ampproject.org/docs/reference/components/amp-pan-zoom)
+    to update the value of transform.scale variable.
+    -->
+    <div>
+     <h1>Seat map</h1>
+     <div class="list-container">
+     <amp-list layout="fill" src="/json/seats.json" items="." single-item noloading>
+      <!--~
+      The seatmap with the availability of the seats is rendered via `amp-list` so that we make sure always the latest
+      data are available to the user.
+      ~-->
+      <template type="amp-mustache">
+         <!--~
+          We are using `layout=fill` to allow the seatmap to use all the available space in the page, this is because
+          we don't know the height or the ratio od the seatmap.
+          ~-->
+        <amp-pan-zoom layout="fill" id="pan-zoom"
+          on="transformEnd:AMP.setState({seatmap: {scale: event.scale, x: event.x, y: event.y, zoom: event.scale <= 1 ? 0 : 1}})">
+          <svg class="seat-map-svg" preserveAspectRatio="xMidYMin slice" viewBox="0 0 {{width}} {{height}}">
+          {{#seats}}                
+          <!--~
+          If a seat is selected, when the user selects it again, it's removed from the seatmap.selectedSeats list
+          ~-->
+            <rect class="seat {{unavailable}}" [class]="seatmap.selectedSeats.includes('{{id}}')? 'seat selected' : 'seat'" 
+              x="{{x}}" y="{{y}}" width="{{width}}" height="{{height}}" role="button" tabindex="0" rx="{{rx}}" ry="{{ry}}"
+              on="{{#available}} tap:AMP.setState({
+                seatmap: {
+                  selectedSeats: !seatmap.selectedSeats.includes('{{id}}') ? 
+                  seatmap.selectedSeats.concat('{{id}}') : copyAndSplice(seatmap.selectedSeats, seatmap.selectedSeats.indexOf('{{id}}'), 1)
+                }
+              }) {{/available}}"></rect>  
+            {{/seats}}
+          </svg>
+        </amp-pan-zoom>
+      </template>
+      <!--~
+      We display a placeholder for the seatmap by using server-side templating. This is possible because we have
+      the information about the layout of the theater.
```

Comment:
Can you rephrase: "To improve the loading experience, we'll show a seatmap skeleton without the actual seat availability. The skeleton is rendered server-side as a `placeholder` element. "

Reply:
done